src/Conversion/ONNXToKrnl/Tensor/Pad.cpp

```diff
@@ -68,11 +69,22 @@ struct ONNXPadOpLowering : public OpConversionPattern<ONNXPadOp> {
       // This way is to avoid using `select` in computing indices as doing for
       // 'edge' and 'reflect' modes.
       Value cValue;
+      SmallVector<Value, 1> loadIndices;
+      if (MemRefType constantValueType =
+              constantValue.getType().dyn_cast<MemRefType>()) {
+        if (constantValueType.getElementType().isF32() &&
+            constantValueType.getRank() == 1) {
+          // If the constant_value type is 1xf32 do krnl.load with an index of
```

Comment:
You can add the issue number in the comment.
Please clearly comment that this fix is a workaround for incorrect model. Could we print a warning message here using llvm warning mechanism?

Reply:
Updated comment.  Tried the emit warning and it caused onnx-mlir --EmitLib to fail.  Thank you.